apps/api/v2/src/modules/organizations/services/event-types/output.service.ts

```diff
@@ -74,25 +77,38 @@ export class OutputOrganizationsEventTypesService {
 
   async getManagedEventTypeHosts(eventTypeId: number) {
     const children = await this.organizationEventTypesRepository.getEventTypeChildren(eventTypeId);
-    const hostsIds: number[] = [];
+    const transformedHosts: TeamEventTypeResponseHost[] = [];
     for (const child of children) {
       if (child.userId) {
-        hostsIds.push(child.userId);
+        const user = await this.usersRepository.findById(child.userId);
+        transformedHosts.push({ userId: child.userId, name: user?.name || "" });
       }
     }
-    return hostsIds.map((userId) => ({ userId }));
+    return transformedHosts;
   }
 
-  transformHosts(hosts: Host[]) {
-    if (!hosts) return [];
+  async transformHosts(
+    hosts: Host[],
+    schedulingType: SchedulingType | null
+  ): Promise<TeamEventTypeResponseHost[]> {
+    if (!schedulingType) return [];
 
-    return hosts.map((host) => {
-      return {
-        userId: host.userId,
-        mandatory: host.isFixed,
-        priority: getPriorityLabel(host.priority || 2),
-      };
-    });
+    const transformedHosts: TeamEventTypeResponseHost[] = [];
+    for (const host of hosts) {
+      const user = await this.usersRepository.findById(host.userId);
+      if (schedulingType === "ROUND_ROBIN") {
+        transformedHosts.push({
+          userId: host.userId,
+          name: user?.name || "",
+          mandatory: host.isFixed,
+          priority: getPriorityLabel(host.priority || 2),
+        });
+      } else {
+        transformedHosts.push({ userId: host.userId, name: user?.name || "" });
+      }
+    }
+
+    return transformedHosts;
```

Comment:
this function could use come comments to explain the logic. also we should first fetch all the required users in one findMany call, we should avoid prisma calls inside of for loops

Reply:
Good call. Implemented in [refactor: transforming response hosts](https://github.com/calcom/cal.com/pull/15928/commits/a81a6408f57eb2cf986ffdbf1f66b26d3cc82657)